### PR TITLE
rgw: test kafka sasl scram

### PR DESCRIPTION
**Pull request by yuvalif**
this PR is intended to test bucket notifications with Kafka security using `SASL` with `SCRAM-SHA-256` mechanism.
following Kafka configuration is used (added to `config/server.properties`):
```
listeners=PLAINTEXT://localhost:9092,SSL://localhost:9093,SASL_SSL://localhost:9094,SASL_PLAINTEXT://localhost:9095
ssl.keystore.location=/home/ylifshit/kafka-3.3.1-src/server.keystore.jks 
ssl.keystore.password=mypassword 
ssl.key.password=mypassword 
ssl.truststore.location=/home/ylifshit/kafka-3.3.1-src/server.truststore.jks 
ssl.truststore.password=mypassword 
sasl.enabled.mechanisms=SCRAM-SHA-256,PLAIN

listener.name.sasl_ssl.scram-sha-256.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required \
   username="alice" \
   password="alice-secret" \
   user_alice="alice-secret";

listener.name.sasl_ssl.plain.sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \
   username="alice" \
   password="alice-secret" \
   user_alice="alice-secret";

listener.name.sasl_plaintext.scram-sha-256.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required \
   username="alice" \
   password="alice-secret" \                                                                                                                                                                                                                  
   user_alice="alice-secret";

listener.name.sasl_plaintext.plain.sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \
   username="alice" \
   password="alice-secret" \
   user_alice="alice-secret";
```

SASL and SSL/SASL tests are passing with `PLAIN` mechanism, but failing with `SCRAM-SHA-256` mechanism:

```
[2022-11-16 15:11:18,342] DEBUG Set SASL server state to HANDSHAKE_OR_VERSIONS_REQUEST during authentication (org.apache.kafka.common.security.authenticator.SaslServerAuthenticator)                                                         
[2022-11-16 15:11:18,342] DEBUG Handling Kafka request API_VERSIONS during authentication (org.apache.kafka.common.security.authenticator.SaslServerAuthenticator)                                                                            
[2022-11-16 15:11:18,342] TRACE [SocketServer listenerType=ZK_BROKER, nodeId=0] clients: Removed a reference to ClientInformation(softwareName=unknown, softwareVersion=unknown).  0 reference(s) remaining. (org.apache.kafka.common.network.Selector)                                                                                                                                                                                                                                     
[2022-11-16 15:11:18,343] DEBUG Set SASL server state to HANDSHAKE_REQUEST during authentication (org.apache.kafka.common.security.authenticator.SaslServerAuthenticator)                                                                     
[2022-11-16 15:11:18,343] DEBUG Handling Kafka request SASL_HANDSHAKE during authentication (org.apache.kafka.common.security.authenticator.SaslServerAuthenticator)                                                                          [2022-11-16 15:11:18,343] DEBUG Using SASL mechanism 'SCRAM-SHA-256' provided by client (org.apache.kafka.common.security.authenticator.SaslServerAuthenticator)                                                                              
[2022-11-16 15:11:18,344] DEBUG Setting SASL/SCRAM_SHA_256 server state to RECEIVE_CLIENT_FIRST_MESSAGE (org.apache.kafka.common.security.scram.internals.ScramSaslServer)                                                                    
[2022-11-16 15:11:18,344] DEBUG Set SASL server state to AUTHENTICATE during authentication (org.apache.kafka.common.security.authenticator.SaslServerAuthenticator)                                                                          [2022-11-16 15:11:18,344] DEBUG Setting SASL/SCRAM_SHA_256 server state to FAILED (org.apache.kafka.common.security.scram.internals.ScramSaslServer)                                                                                          
[2022-11-16 15:11:18,344] DEBUG Set SASL server state to FAILED during authentication (org.apache.kafka.common.security.authenticator.SaslServerAuthenticator)    
```

> Note: DO NOT MERGE. the only new commit in this PR is: ddf48147cfb2ab7c112629ed038f518f1144a07d
all other commits exists in other PRs

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/radosgw/notifications.rst
Comment (anthonyeleven) [inline doc/radosgw/notifications.rst:216]: Period at the end of lines 215 and 216?

File: src/test/rgw/bucket_notification/README.rst
Comment (anthonyeleven) [inline src/test/rgw/bucket_notification/README.rst:52]: Backticks around `vstart`? Maybe `vstart.sh` as well?

File: src/test/rgw/bucket_notification/README.rst
Comment (anthonyeleven) [inline src/test/rgw/bucket_notification/README.rst:102]: Kafka spelling

**Review by anthonyeleven — CHANGES_REQUESTED** [review]
see comments

**Review by anthonyeleven — APPROVED** [review]

**Comment by anthonyeleven** [comment]
Docs LGTM modulo the Sphinx errors.

**Comment by yuvalif** [comment]
passing teuthology: http://pulpito.front.sepia.ceph.com/yuvalif-2023-01-21_16:07:53-rgw:notifications-test-kafka-sasl-scram-distro-default-smithi/

once PR #48181 is merged, the test commit from this PR (8339be11fbea9578df4b99ded7ebafcf14d8a653) should be merged as well

**Comment by yuvalif** [comment]
jenkins test make check

**Comment by yuvalif** [comment]
@TRYTOBE8TME  @huynp1999 can you please take a look?

**Comment by yuvalif** [comment]
teuthology results: http://pulpito.front.sepia.ceph.com/yuvalif-2023-03-07_12:11:46-rgw:notifications-wip-yuval-test1-distro-default-smithi/

**Comment by yuvalif** [comment]
teuthology is passing: http://pulpito.front.sepia.ceph.com/yuvalif-2023-03-07_12:11:46-rgw:notifications-wip-yuval-test1-distro-default-smithi/

**Review by cbodley — APPROVED** [review]

**Pull request merged**
